src/irmin-pack/unix/append_only_file.ml

```diff
@@ -52,19 +52,31 @@ module Make (Io : Io.S) = struct
       [open_rw] functions. The [end_offset] from the control file is then used
       as the real offset.
 
-      In case of a crash, if the real offset is larger than the [end_offset], we
-      can safely revert back to [end_offset]. If not, then the store is
-      corrupted, an error is returned. *)
-  let check_consistent_store ~end_offset io =
+      In case of a crash:
+
+      - if the real offset is larger than the [end_offset], we can safely revert
+        back to [end_offset].
```

Comment:
this is actually not what we discussed with @samoht and @metanivek - we should always take the real_offset and update the control file accordingly. In this way, we ensure that we don't crash if the index points to something between [end_offset] and the real offset. I'll push a fix.